tests/test_regrid.py

```diff
@@ -195,3 +195,35 @@ def test_out_of_bounds():
     output = regrid(data)
 
     assert torch.all(torch.isnan(output))
+
+
+def test_NearestNeighborInterpolator():
+    n = 10000
+    torch.manual_seed(0)
+    lon = torch.rand(n) * 360
+    lat = torch.rand(n) * 180 - 90
+
+    lond = torch.rand(n) * 360
+    latd = torch.rand(n) * 180 - 90
+
+    interpolate = earth2grid.S2NearestNeighborInterpolator(lon, lat, lond, latd)
+    out = interpolate(torch.cos(torch.deg2rad(lon)))
+    expected = torch.cos(torch.deg2rad(lond))
+    mae = torch.mean(torch.abs(out - expected))
+    assert mae.item() < 0.02
+
+
+def test_BaryCentric():
+    n = 10000
+    torch.manual_seed(0)
+    lon = torch.rand(n) * 360
+    lat = torch.rand(n) * 180 - 90
```

Comment:
```suggestion
    lat = torch.asin(2*torch.rand(n)-1) * 180. / np.pi
```

Reply:
what is the intent here?

Reply:
This isn't a super important change, but thought it would be good to remove the polar bias when sampling on the sphere in these tests.